server/core/src/main/java/org/infinispan/server/core/dataconversion/deserializer/Deserializer.java

```diff
@@ -0,0 +1,603 @@
+package org.infinispan.server.core.dataconversion.deserializer;
+
+import static java.io.ObjectStreamConstants.SC_BLOCK_DATA;
+import static java.io.ObjectStreamConstants.SC_EXTERNALIZABLE;
+import static java.io.ObjectStreamConstants.SC_SERIALIZABLE;
+import static java.io.ObjectStreamConstants.SC_WRITE_METHOD;
+import static java.io.ObjectStreamConstants.STREAM_MAGIC;
+import static java.io.ObjectStreamConstants.STREAM_VERSION;
+import static java.io.ObjectStreamConstants.TC_ARRAY;
+import static java.io.ObjectStreamConstants.TC_BLOCKDATA;
+import static java.io.ObjectStreamConstants.TC_BLOCKDATALONG;
+import static java.io.ObjectStreamConstants.TC_CLASS;
+import static java.io.ObjectStreamConstants.TC_CLASSDESC;
+import static java.io.ObjectStreamConstants.TC_ENDBLOCKDATA;
+import static java.io.ObjectStreamConstants.TC_ENUM;
+import static java.io.ObjectStreamConstants.TC_EXCEPTION;
+import static java.io.ObjectStreamConstants.TC_LONGSTRING;
+import static java.io.ObjectStreamConstants.TC_NULL;
+import static java.io.ObjectStreamConstants.TC_OBJECT;
+import static java.io.ObjectStreamConstants.TC_PROXYCLASSDESC;
+import static java.io.ObjectStreamConstants.TC_REFERENCE;
+import static java.io.ObjectStreamConstants.TC_RESET;
+import static java.io.ObjectStreamConstants.TC_STRING;
+import static java.io.ObjectStreamConstants.baseWireHandle;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.io.StreamCorruptedException;
+import java.io.WriteAbortedException;
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+
+/**
+ * Based on Serialisys by Eamonn McManus
+ */
+public class Deserializer {
+
+   private static final SEntity END = new SString("END");
+   private final PrimitiveClassDescFactory primitiveClassDescFactory = new PrimitiveClassDescFactory();
+   private final DataInputStream din;
+   private final List<SEntity> handles = new ArrayList<>();
+   private final Map<String, ClassDesc> classDescriptions = new HashMap<>();
+
+   public Deserializer(InputStream in, boolean header) throws IOException {
+      this.din = new DataInputStream(in);
```

Comment:
Just to confirm.  We're not vulnerable to Java serialization exploits here, and therefore we don't need to respect the configured allow-list, as we're not reading via an `ObjectInputStream` right?

Reply:
Exactly: this never instantiates any classes (aside from strings)